kernel/typeops.ml

```diff
@@ -431,7 +431,39 @@ and execute_recdef env (names,lar,vdef) i =
 and execute_array env = Array.map (execute env)
 
 (* Derived functions *)
+
+let universe_levels_of_constr env c =
+  let open Declarations in
+  let rec aux s c =
+    match kind c with
+    | Const (c, u) ->
+      begin match (Environ.lookup_constant c env).const_universes with
+        | Polymorphic_const _ ->
+          LSet.fold LSet.add (Instance.levels u) s
+        | Monomorphic_const (univs, _) ->
+          LSet.union s univs
```

Comment:
I know this is just moving code, but it's fishy to add the global universes to the one appearing in the term. They're not quite part of the term, sometimes they're even half garbage (e.g. side-effects). It's fine because it's an overapproximation, but still.